### ISSUE TITLE
Trying to run MiniGPT-4 locally but component called "LLAMA" appears to be missing.

### ISSUE THREAD

**Issue by PeterSwinkels** [Vision-CAIR/MiniGPT-4#239]
(Apologies for the clumsy formatting.)

Hello,

I am trying to run MiniGPT-4 locally on my PC (Windows 10).

After installing Python 3.11.3 (64-bit) I downloaded the code for MiniGPT-4 at https://github.com/Vision-CAIR/MiniGPT-4.

After that I ran the following commands to install all dependencies I am aware of:
---
C:\Users\...\AppData\Local\Programs\Python\Python311\python.exe -m pip install --upgrade pip
pip install numpy
pip install torch
pip install opencv-python
pip install gradio
pip install omegaconf
pip install torchvision
pip install iopath
pip install timm
pip install webdataset
pip install transformers
pip install decord
pip install sentencepiece
---

When all that has been done I get the folllowing error when trying to launch MiniGPT-4 using "C:\Users\...\AppData\Local\Programs\Python\Python311\python demo.py --cfg-path eval_configs/minigpt4_eval.yaml  --gpu-id 0":

---
Initializing Chat
Loading VIT
Loading VIT Done
Loading Q-Former
Loading Q-Former Done
Loading LLAMA
Traceback (most recent call last):
  File "D:\Other\MiniGPT-4-main\demo.py", line 60, in <module>
    model = model_cls.from_config(model_config).to('cuda:{}'.format(args.gpu_id))
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Other\MiniGPT-4-main\minigpt4\models\mini_gpt4.py", line 243, in from_config
    model = cls(
            ^^^^
  File "D:\Other\MiniGPT-4-main\minigpt4\models\mini_gpt4.py", line 86, in __init__
    self.llama_tokenizer = LlamaTokenizer.from_pretrained(llama_model, use_fast=False)
                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\peter\AppData\Local\Programs\Python\Python311\Lib\site-packages\transformers\tokenization_utils_base.py", line 1771, in from_pretrained
    resolved_vocab_files[file_id] = cached_file(
                                    ^^^^^^^^^^^^
  File "C:\Users\peter\AppData\Local\Programs\Python\Python311\Lib\site-packages\transformers\utils\hub.py", line 417, in cached_file
    resolved_file = hf_hub_download(
                    ^^^^^^^^^^^^^^^^
  File "C:\Users\peter\AppData\Local\Programs\Python\Python311\Lib\site-packages\huggingface_hub\utils\_validators.py", line 112, in _inner_fn
    validate_repo_id(arg_value)
  File "C:\Users\peter\AppData\Local\Programs\Python\Python311\Lib\site-packages\huggingface_hub\utils\_validators.py", line 160, in validate_repo_id
    raise HFValidationError(
huggingface_hub.utils._validators.HFValidationError: Repo id must be in the form 'repo_name' or 'namespace/repo_name': '/path/to/vicuna/weights/'. Use `repo_type` argument if needed.
D:\Other\MiniGPT-4-main>
---

Searching MiniGPT-4.pdf for the term "LLAMA" leads me to this paragraph:
---
[30] Rohan Taori, Ishaan Gulrajani, Tianyi Zhang, Yann Dubois, Xuechen Li, Carlos Guestrin, Percy Liang,
and Tatsunori B. Hashimoto. Stanford alpaca: An instruction-following llama model. https://github.
com/tatsu-lab/stanford_alpaca, 2023.
---

Does anyone know whether this "stanford-alpaca" is what is missing or whether it is what "LLAMA" is based on?

In short what dependency is missing and how do I install it?

I truly think machine vision and ai capable chat systems are  a fascinating and important development. Any help in getting MiniGPT-4 to run locally on my machine will be greatly appreciated! Thanks in advance.

**Comment by ttio2tech** [comment]
You need to specify the path to the llama_model in [eval_configs/minigpt4_eval.yaml](https://github.com/Vision-CAIR/MiniGPT-4/blob/main/minigpt4/configs/models/minigpt4.yaml#L16)

**Comment by PeterSwinkels** [comment]
@ttio2tech: Thank you for your help. How do I get the llama model?

**Comment by ttio2tech** [comment]
> @ttio2tech: Thank you for your help. How do I get the llama model?

There are instructions on the README (Installation, prepare VICUNA...)

**Comment by PeterSwinkels** [comment]
Found it. Git appears to be downloading vicuna atm. How many gbs of diskspace does it need?

**Comment by PeterSwinkels** [comment]
I followed the steps to install vicuna:

D:\Other\MiniGPT-4-main>python demo.py --cfg-path eval_configs/minigpt4_eval.yaml  --gpu-id 0
Initializing Chat
Loading VIT
Loading VIT Done
Loading Q-Former
Loading Q-Former Done
Loading LLAMA

===================================BUG REPORT===================================
Welcome to bitsandbytes. For bug reports, please run

python -m bitsandbytes

 and submit this information together with your error trace to: https://github.com/TimDettmers/bitsandbytes/issues
================================================================================
bin C:\Users\peter\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.10_qbz5n2kfra8p0\LocalCache\local-packages\Python310\site-packages\bitsandbytes\libbitsandbytes_cpu.so
C:\Users\peter\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.10_qbz5n2kfra8p0\LocalCache\local-packages\Python310\site-packages\bitsandbytes\cextension.py:33: UserWarning: The installed version of bitsandbytes was compiled without GPU support. 8-bit optimizers, 8-bit multiplication, and GPU quantization are unavailable.
  warn("The installed version of bitsandbytes was compiled without GPU support. "
CUDA SETUP: Loading binary C:\Users\peter\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.10_qbz5n2kfra8p0\LocalCache\local-packages\Python310\site-packages\bitsandbytes\libbitsandbytes_cpu.so...
argument of type 'WindowsPath' is not iterable
Loading checkpoint shards:   0%|                                                                 | 0/3 [00:11<?, ?it/s]
╭─────────────────────────────── Traceback (most recent call last) ────────────────────────────────╮
│ D:\Other\MiniGPT-4-main\demo.py:60 in <module>                                                   │
│                                                                                                  │
│    57 model_config = cfg.model_cfg                                                               │
│    58 model_config.device_8bit = args.gpu_id                                                     │
│    59 model_cls = registry.get_model_class(model_config.arch)                                    │
│ ❱  60 model = model_cls.from_config(model_config).to('cuda:{}'.format(args.gpu_id))              │
│    61                                                                                            │
│    62 vis_processor_cfg = cfg.datasets_cfg.cc_sbu_align.vis_processor.train                      │
│    63 vis_processor = registry.get_processor_class(vis_processor_cfg.name).from_config(vis_pro   │
│                                                                                                  │
│ D:\Other\MiniGPT-4-main\minigpt4\models\mini_gpt4.py:243 in from_config                          │
│                                                                                                  │
│   240 │   │   max_txt_len = cfg.get("max_txt_len", 32)                                           │
│   241 │   │   end_sym = cfg.get("end_sym", '\n')                                                 │
│   242 │   │                                                                                      │
│ ❱ 243 │   │   model = cls(                                                                       │
│   244 │   │   │   vit_model=vit_model,                                                           │
│   245 │   │   │   q_former_model=q_former_model,                                                 │
│   246 │   │   │   img_size=img_size,                                                             │
│                                                                                                  │
│ D:\Other\MiniGPT-4-main\minigpt4\models\mini_gpt4.py:90 in __init__                              │
│                                                                                                  │
│    87 │   │   self.llama_tokenizer.pad_token = self.llama_tokenizer.eos_token                    │
│    88 │   │                                                                                      │
│    89 │   │   if self.low_resource:                                                              │
│ ❱  90 │   │   │   self.llama_model = LlamaForCausalLM.from_pretrained(                           │
│    91 │   │   │   │   llama_model,                                                               │
│    92 │   │   │   │   torch_dtype=torch.float16,                                                 │
│    93 │   │   │   │   load_in_8bit=True,                                                         │
│                                                                                                  │
│ C:\Users\peter\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.10_qbz5n2kfra8p0\LocalCa │
│ che\local-packages\Python310\site-packages\transformers\modeling_utils.py:2784 in                │
│ from_pretrained                                                                                  │
│                                                                                                  │
│   2781 │   │   │   │   mismatched_keys,                                                          │
│   2782 │   │   │   │   offload_index,                                                            │
│   2783 │   │   │   │   error_msgs,                                                               │
│ ❱ 2784 │   │   │   ) = cls._load_pretrained_model(                                               │
│   2785 │   │   │   │   model,                                                                    │
│   2786 │   │   │   │   state_dict,                                                               │
│   2787 │   │   │   │   loaded_state_dict_keys,  # XXX: rename?                                   │
│                                                                                                  │
│ C:\Users\peter\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.10_qbz5n2kfra8p0\LocalCa │
│ che\local-packages\Python310\site-packages\transformers\modeling_utils.py:3125 in                │
│ _load_pretrained_model                                                                           │
│                                                                                                  │
│   3122 │   │   │   │   )                                                                         │
│   3123 │   │   │   │                                                                             │
│   3124 │   │   │   │   if low_cpu_mem_usage:                                                     │
│ ❱ 3125 │   │   │   │   │   new_error_msgs, offload_index, state_dict_index = _load_state_dict_i  │
│   3126 │   │   │   │   │   │   model_to_load,                                                    │
│   3127 │   │   │   │   │   │   state_dict,                                                       │
│   3128 │   │   │   │   │   │   loaded_keys,                                                      │
│                                                                                                  │
│ C:\Users\peter\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.10_qbz5n2kfra8p0\LocalCa │
│ che\local-packages\Python310\site-packages\transformers\modeling_utils.py:717 in                 │
│ _load_state_dict_into_meta_model                                                                 │
│                                                                                                  │
│    714 │   │   │   │   fp16_statistics = None                                                    │
│    715 │   │   │                                                                                 │
│    716 │   │   │   if "SCB" not in param_name:                                                   │
│ ❱  717 │   │   │   │   set_module_8bit_tensor_to_device(                                         │
│    718 │   │   │   │   │   model, param_name, param_device, value=param, fp16_statistics=fp16_s  │
│    719 │   │   │   │   )                                                                         │
│    720                                                                                           │
│                                                                                                  │
│ C:\Users\peter\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.10_qbz5n2kfra8p0\LocalCa │
│ che\local-packages\Python310\site-packages\transformers\utils\bitsandbytes.py:87 in              │
│ set_module_8bit_tensor_to_device                                                                 │
│                                                                                                  │
│    84 │   │   if value is None:                                                                  │
│    85 │   │   │   new_value = old_value.to(device)                                               │
│    86 │   │   elif isinstance(value, torch.Tensor):                                              │
│ ❱  87 │   │   │   new_value = value.to(device)                                                   │
│    88 │   │   else:                                                                              │
│    89 │   │   │   new_value = torch.tensor(value, device=device)                                 │
│    90                                                                                            │
│                                                                                                  │
│ C:\Users\peter\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.10_qbz5n2kfra8p0\LocalCa │
│ che\local-packages\Python310\site-packages\torch\cuda\__init__.py:239 in _lazy_init              │
│                                                                                                  │
│    236 │   │   │   │   "Cannot re-initialize CUDA in forked subprocess. To use CUDA with "       │
│    237 │   │   │   │   "multiprocessing, you must use the 'spawn' start method")                 │
│    238 │   │   if not hasattr(torch._C, '_cuda_getDeviceCount'):                                 │
│ ❱  239 │   │   │   raise AssertionError("Torch not compiled with CUDA enabled")                  │
│    240 │   │   if _cudart is None:                                                               │
│    241 │   │   │   raise AssertionError(                                                         │
│    242 │   │   │   │   "libcudart functions unavailable. It looks like you have a broken build?  │
╰──────────────────────────────────────────────────────────────────────────────────────────────────╯
AssertionError: Torch not compiled with CUDA enabled

D:\Other\MiniGPT-4-main>

**Comment by PeterSwinkels** [comment]
This thing is impossible to install. One barrage of error messages after the other. I am closing this issue.

**Comment by ttio2tech** [comment]
Seems your Pytorch install has issues. 
I was able to self host the 7B version at [agireact.com](https://agireact.com/)
Can also access it via discord bot. [Server invite](https://discord.com/invite/SgmBydQ2Mn)
Feel free to give it a try.
